test/desugar/call_indirect.txt

```diff
@@ -0,0 +1,24 @@
+;;; TOOL: wat-desugar
```

Comment:
Does this issue only effect wat-desugar?  From the change it looks like it would also effect wasm2wat?

Reply:
I think the BinaryReaderIR is never going to make a CallIndirect Expr where decl.has_func_type is false, because the binary format always has a sig_index right there physically. So I think the only way this can happen is if the Module originated as WAT and is being written out again as WAT. (Which afaik only happens in wat-desugar.) FWIW the BinaryWriter always does this lookup unconditionally.

Reply:
I wonder if we should just drop `wat-desugar` completely.. and have folks use `wat2wasm + wasm2wat` to get this effect?    Is this tool worth maintaining on its own?

Reply:
No objection from me to getting rid of wat-desugar. I doubt it gets a lot of use. But we should probably fix this WatWriter issue either way. :-)

Reply:
Sure this lgtm either way.